src/main/java/frc/robot/commandgroups/OneButtonAdjustAndShoot.java

```diff
@@ -0,0 +1,36 @@
+package frc.robot.commandgroups;
+
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Constants;
+import frc.robot.RobotContainer;
+import frc.robot.subsystems.conveyor.Conveyor;
+import frc.robot.subsystems.conveyor.commands.Convey;
+import frc.robot.subsystems.drivetrain.SwerveDrive;
+import frc.robot.subsystems.flap.Flap;
+import frc.robot.subsystems.hood.Hood;
+
+public class OneButtonAdjustAndShoot extends SequentialCommandGroup {
+    public OneButtonAdjustAndShoot(SwerveDrive swerve, Conveyor conveyor, Flap flap, Hood hood) {
+        addCommands(new InstantCommand(flap::allowShooting));
+
+        addCommands(new ParallelCommandGroup(
+                new Convey(conveyor, -0.25).withTimeout(0.075),
+                //if (!RobotContainer.hasTarget.getAsBoolean())
+                new OdometryAdjust(swerve)
```

Comment:
What happens if the odometry is incorrect? Is there some kind of way to overrule this

Reply:
I don't think the odometry could ever get so bad to the point it's unusable. I was thinking to use odometry only when you don't see the target but I'm thinking this will be more accurate. We could put a button that does that. In any way, after odometry comes the vision fix which should make it reliable either way.

Reply:
Okay we'll have to see it in drivers practice but it might be useful to split it a bit.